File: backend/routes/clients.py
from fastapi import APIRouter, HTTPException, Request
from datetime import datetime
from bson import ObjectId

router = APIRouter()

async def get_client_stats(db):
    total = await db.client_companies.count_documents({})
    active = await db.client_companies.count_documents({"isActive": True})
    return {"total": total, "active": active}

@router.get("/")
async def get_clients(request: Request, limit: int = 50, offset: int = 0):
    db = request.app.state.db
    
    cursor = db.client_companies.find({}).sort("createdAt", -1).skip(offset).limit(limit)
    clients = await cursor.to_list(length=limit)
    total = await db.client_companies.count_documents({})
    
    for client in clients:
        client["id"] = str(client.pop("_id"))
    
    return {"clients": clients, "total": total}

@router.get("/{client_id}")
async def get_client(client_id: str, request: Request):
    db = request.app.state.db
    
    client = await db.client_companies.find_one({"_id": ObjectId(client_id)})
    if not client:
        raise HTTPException(status_code=404, detail="Client not found")
    
    client["id"] = str(client.pop("_id"))
    return client

@router.post("/")
async def create_client(request: Request):
    db = request.app.state.db
    data = await request.json()
    
    client_doc = {
        "name": data.get("name"),
        "email": data.get("email"),
        "phone": data.get("phone"),
        "industry": data.get("industry"),
        "subscription": data.get("subscription", "basic"),
        "services": data.get("services", []),
        "isActive": data.get("isActive", True),
        "createdAt": datetime.utcnow()
    }
    
    result = await db.client_companies.insert_one(client_doc)
    client_doc["id"] = str(result.inserted_id)
    
    return {"message": "Client created", "client": client_doc}

@router.put("/{client_id}")
async def update_client(client_id: str, request: Request):
    db = request.app.state.db
    data = await request.json()
    
    update_data = {k: v for k, v in data.items() if k not in ["_id", "id"]}
    update_data["updatedAt"] = datetime.utcnow()
    
    result = await db.client_companies.update_one(
        {"_id": ObjectId(client_id)},
        {"$set": update_data}
    )
    
    if result.matched_count == 0:
        raise HTTPException(status_code=404, detail="Client not found")
    
    return {"message": "Client updated"}

@router.delete("/{client_id}")
async def delete_client(client_id: str, request: Request):
    db = request.app.state.db
    
    result = await db.client_companies.delete_one({"_id": ObjectId(client_id)})
    
    if result.deleted_count == 0:
        raise HTTPException(status_code=404, detail="Client not found")
    
    return {"message": "Client deleted"}

File: backend/routes/clients.js
const express = require('express');
const { body, validationResult } = require('express-validator');
const ClientCompany = require('../models/ClientCompany');
const { verifyAdminToken } = require('./adminAuth');
const router = express.Router();

// Get all clients (admin)
router.get('/', verifyAdminToken, async (req, res) => {
  try {
    const { status, plan, industry, search } = req.query;
    const query = {};
    if (status) query['subscription.status'] = status;
    if (plan) query['subscription.plan'] = plan;
    if (industry) query.industry = industry;
    if (search) {
      query.$or = [
        { name: { $regex: search, $options: 'i' } },
        { email: { $regex: search, $options: 'i' } }
      ];
    }
    const clients = await ClientCompany.find(query).sort({ createdAt: -1 });
    res.json({ clients });
  } catch (error) {
    res.status(500).json({ message: 'Internal server error' });
  }
});

// Get single client (admin)
router.get('/:id', verifyAdminToken, async (req, res) => {
  try {
    const client = await ClientCompany.findById(req.params.id);
    if (!client) return res.status(404).json({ message: 'Client not found' });
    res.json({ client });
  } catch (error) {
    res.status(500).json({ message: 'Internal server error' });
  }
});

// Create client (admin)
router.post('/', verifyAdminToken, [
  body('name').trim().notEmpty(),
  body('email').isEmail().normalizeEmail()
], async (req, res) => {
  try {
    const errors = validationResult(req);
    if (!errors.isEmpty()) {
      return res.status(400).json({ message: 'Validation failed', errors: errors.array() });
    }
    const existingClient = await ClientCompany.findOne({ email: req.body.email });
    if (existingClient) {
      return res.status(409).json({ message: 'Client already exists' });
    }
    const client = new ClientCompany(req.body);
    await client.save();
    res.status(201).json({ message: 'Client created', client });
  } catch (error) {
    res.status(500).json({ message: 'Internal server error' });
  }
});

// Update client (admin)
router.put('/:id', verifyAdminToken, async (req, res) => {
  try {
    const client = await ClientCompany.findByIdAndUpdate(req.params.id, req.body, { new: true });
    if (!client) return res.status(404).json({ message: 'Client not found' });
    res.json({ message: 'Client updated', client });
  } catch (error) {
    res.status(500).json({ message: 'Internal server error' });
  }
});

// Delete client (admin)
router.delete('/:id', verifyAdminToken, async (req, res) => {
  try {
    await ClientCompany.findByIdAndDelete(req.params.id);
    res.json({ message: 'Client deleted' });
  } catch (error) {
    res.status(500).json({ message: 'Internal server error' });
  }
});

// Get stats (admin)
router.get('/stats/overview', verifyAdminToken, async (req, res) => {
  try {
    const [total, active, byPlan, byIndustry] = await Promise.all([
      ClientCompany.countDocuments(),
      ClientCompany.countDocuments({ isActive: true, 'subscription.status': 'active' }),
      ClientCompany.aggregate([{ $group: { _id: '$subscription.plan', count: { $sum: 1 } } }]),
      ClientCompany.aggregate([{ $group: { _id: '$industry', count: { $sum: 1 } } }])
    ]);
    res.json({
      stats: {
        total, active,
        byPlan: byPlan.reduce((acc, curr) => { acc[curr._id || 'none'] = curr.count; return acc; }, {}),
        byIndustry: byIndustry.reduce((acc, curr) => { acc[curr._id || 'none'] = curr.count; return acc; }, {})
      }
    });
  } catch (error) {
    res.status(500).json({ message: 'Internal server error' });
  }
});

module.exports = router;
